system-addon/lib/TopSitesFeed.jsm

```diff
@@ -190,6 +201,31 @@ this.TopSitesFeed = class TopSitesFeed {
     }
   }
 
+/**
+ * Fetch, cache and broadcast a custom screenshot for a specific topsite.
+ * @param link cached topsite object
+ */
+  async _fetchCustomScreenshot(link) {
+    await Screenshots.maybeCacheScreenshot(link, link.customScreenshotURL, "customScreenshot",
+      customScreenshot => this.store.dispatch(ac.BroadcastToContent({
+        data: {customScreenshot, url: link.url},
+        type: at.SCREENSHOT_UPDATED
```

Comment:
This seems quite similar to the screenshot request from `_fetchIcon`, so maybe worth combining unless they actually should be doing something different.

Reply:
created `_fetchScreenshot` used to get the custom urls and the regular urls (in `_fetchIcon`)